Problems 7,8,10,15,19

In [45]:
%matplotlib inline

import numpy as np
import scipy.stats as st
import pandas as pd
import statsmodels.api as sm
import statsmodels.stats.api as sms
import matplotlib.pyplot as plt
import seaborn as sns
from IPython.display import Image
from statsmodels import stats

## 7.7

### a.

In [46]:
# Load the commercial properties data
cp_df = pd.read_table('/Users/kevin/Dropbox/School/STA-580/ch6hw/CH06PR18.txt',
                     sep='\s*', index_col=False, engine='python',
                     names=['Y-rental_rate', 'X1-age', 'X2-expenses', 'X3-vacancy_rate', 'X4-footage'])
cp_df.head()

Y-rental_rate  X1-age  X2-expenses  X3-vacancy_rate  X4-footage
0           13.5       1         5.02             0.14      123000
1           12.0      14         8.19             0.27      104079
2           10.5      16         3.00             0.00       39998
3           15.0       4        10.70             0.05       57112
4           14.0      11         8.97             0.07       60000

The desired regression sums of squares (associated with X4, X1 given X4, etc.) are in the "Seq SS" column:

    Analysis of Variance

    Source               DF   Seq SS  Contribution   Adj SS   Adj MS  F-Value  P-Value
    Regression            4  138.327        58.47%  138.327  34.5817    26.76    0.000
      X4 - footage        1   67.775        28.65%   42.325  42.3250    32.75    0.000
      X1 - age            1   42.275        17.87%   57.243  57.2428    44.29    0.000
      X2 - expenses       1   27.857        11.78%   25.759  25.7590    19.93    0.000
      X3 - vacancy rate   1    0.420         0.18%    0.420   0.4197     0.32    0.570
    Error                76   98.231        41.53%   98.231   1.2925
    Total                80  236.558       100.00%
    
To check this, I'll regress Y on X4:

    Source          DF   Seq SS  Contribution   Adj SS   Adj MS  F-Value  P-Value
    Regression       1   67.775        28.65%   67.775  67.7751    31.72    0.000
    
And then Y on X4 and X1:

    Source          DF   Seq SS  Contribution   Adj SS   Adj MS  F-Value  P-Value
    Regression       2  110.050        46.52%  110.050  55.0248    33.93    0.000
    
Both approaches agree that $SSR(X_4) = 67.775$. And:

$$SSR(X_1 \mid X_4) = SSR(X_1,X_4) - SSR(X_4) = 110.050 - 67.775 = 42.275$$

, which agrees with the value in the Seq SS column above as well. 

### b.

The alternatives are:

$$H_0: \beta_3 = 0$$
$$H_a: \beta_3 \ne 0$$

If $H_0$ holds, $F^* \sim F(1,n-p)$. Large values of $F^*$ lead to $H_a$. 

You can see from the ANOVA table above that $F^*$ = 0.32 and $P$-value = 0.570. We conclude $H_0$ ($X_3$ can be dropped from the regression model given that $X_1$, $X_2$ and $X_4$ are retained). 

## 7.8

The alternatives are:

$$
\begin{aligned}
H_0&: \beta_2 = \beta_3 = 0 \\
H_a&: \text{not both } \beta_2 \text{ and } \beta_3 \text{ equal } 0 \\
\end{aligned}
$$

If $H_0$ holds, $F^* \sim F(p-q,n-p)$. Large values of $F^*$ lead to $H_a$. 

Using (7.27) and (7.28):

$$F^* = \frac{SSR(X_2 \mid X_1,X_3,X_4)+SSR(X_3 \mid X_1,X_2,X_4)}{p-q} \div \frac{SSE(X_1,X_2,X_3,X_4)}{n-p}$$

Substituting from the ANOVA table above, we have

$$F^* = \frac{25.7590+0.4197}{5-3} \div \frac{98.231}{81-5}$$

In [47]:
f_star_8 = (25.7590+0.4197)/(5-3)/(98.231/(81-5))
f_star_8

10.127053577791125

$F(p-q,n-p)  = F(0.01,2,76)$:

In [48]:
f_crit_8 = st.f.ppf(0.99,2,76)
f_crit_8

4.89583988401818

Since $F^*$ > $F_{critical}$, we conclude $H_a$ (we cannot drop both $X_2$ and $X_3$ from the regression model). The $P$-value is:

In [49]:
st.f.sf(f_star_8, 2,76)

0.00012617817769108752

## 7.10

Alternatives:

$$
\begin{aligned}
H_0&: \beta_1 = -0.1, \beta_2 = 0.4 \\
H_a&: \text{not both equalities in } H_0 \text{ hold} \\
\end{aligned}
$$

Full model:

$$Y_i = \beta_0 + \beta_1X_{i1} + \beta_2X_{i2} + \beta_3X_{i3} + \beta_4X_{i4}$$

$$SSE(F) = 98.231$$

$$df_F = 76$$

Reduced model:

$$Y_i + 0.1X_{i1} - 0.4X_{i2} = \beta_0 + \beta_3X_{i3} + \beta_4X_{i4}$$

$$SSE(R) = 110.141$$

$$df_R = 78$$

Using (2.70):

$$F^* = \frac{110.141-98.231}{78-76} \div \frac{98.231}{76}$$

In [50]:
f_star_10 = (110.141-98.231)/(78-76)/98.231/76
f_star_10

0.0007976632952723154

The decision rule is in (2.71).

In [51]:
f_crit_10 = st.f.ppf(1-0.01,78-76,76)
f_crit_10

4.89583988401818

Since $F^* < F_{crit}$, we conclude $H_0$ (there is not evidence to support both equalites not being true).

## 7.15

We'll be using (7.35)-(7.40). We regress $Y$ on $X_4$:

    Regression Analysis: Y - rental rate versus X4 - footage 

          S    R-sq  R-sq(adj)    PRESS  R-sq(pred)
    1.46167  28.65%     27.75%  178.578      24.51%

$$R^2_{Y4} = 0.2865$$

$R^2_{Y4}$ is the percentage of variation in $Y$ explained by $X_4$.

We regress $Y$ on $X_1$:

    Regression Analysis: Y - rental rate versus X1 - age 

          S   R-sq  R-sq(adj)    PRESS  R-sq(pred)
    1.67536  6.26%      5.08%  233.922       1.11%

$$R^2_{Y1} = 0.0626$$

$R^2_{Y1}$ is the percentage of variation in $Y$ explained by $X_1$.

We take $SSR(X_1 \mid X_4)$ from the ANOVA table above and $SSE(X_4)$ from regressing $Y$ on $X_4$:

    Regression Analysis: Y - rental rate versus X4 - footage 

    Source          DF   Seq SS  Contribution   Adj SS   Adj MS  F-Value  P-Value
    Error           79  168.782        71.35%  168.782   2.1365

$$R^2_{Y1 \mid 4} = \frac{SSR(X_1 \mid X_4)}{SSE(X_4)} = \frac{42.275}{168.782} = 0.2505 $$

(See below for calculations.)

$R^2_{Y1 \mid 4}$ is the percentage of variation in $Y$ explained by $X_1$ given that $X_4$ is already in the model. Or, in the language of Kutner et al., it's the proportionate reduction in the variation in $Y$ remaining after $X_4$ is included in the model that is gained by also including $X_1$ in the model.

We regress $Y$ on $X_1$ and $X_4$:

    Regression Analysis: Y - rental rate versus X4 - footage, X1 - age 

          S    R-sq  R-sq(adj)    PRESS  R-sq(pred)
    1.27354  46.52%     45.15%  138.412      41.49%

$$R^2_{Y14} = 0.4652$$

$R^2_{Y14}$ is the variation in Y explained by $X_1$ and $X_4$ together. 

We take $SSR(X_2 \mid X_1,X_4)$ from the ANOVA table above and $SSE(X_1,X_4)$ from regressing $Y$ on $X_1$ and $X_4$:

    Regression Analysis: Y - rental rate versus X4 - footage, X1 - age 

    Source          DF   Seq SS  Contribution   Adj SS   Adj MS  F-Value  P-Value
    Error           78  126.508        53.48%  126.508   1.6219

$$R^2_{Y2 \mid 14} = \frac{SSR(X_2 \mid X_1,X_4)}{SSE(X_1,X_4)} = \frac{27.857}{126.508} = 0.2202$$

$R^2_{Y2 \mid 14}$ is the proportionate reduction in the variation in $Y$ from adding $X_2$ to the model, given that $X_1$ and $X_4$ are already in the model.

We take $SSR(X_3 \mid X_1,X_2,X_4)$ from the ANOVA table above and $SSE(X_1,X_2,X_4)$ from regressing $Y$ on $X_1$, $X_2$, and $X_4$:

    Regression Analysis: Y - rental rate versus X1 - age, X2 - expenses, X4 - footage 

    Source           DF  Seq SS  Contribution  Adj SS  Adj MS  F-Value  P-Value
    Error            77   98.65        41.70%   98.65   1.281

$$R^2_{Y3 \mid 124} = \frac{SSR(X_3 \mid X_1,X_2,X_4)}{SSE(X_1,X_2,X_4)} = \frac{0.420}{98.65} = 0.0043$$

$R^2_{Y3 \mid 124}$ is the proportionate reduction in the variation in $Y$ from adding $X_3$ to the model, given that $X_1$, $X_2$ and $X_4$ are already in the model.

$R^2$ comes from the full model:

    Regression Analysis: Y - rental r versus X4 - footage, X1 - age, X2 - expense, X3 - vacancy 

          S    R-sq  R-sq(adj)    PRESS  R-sq(pred)
    1.13689  58.47%     56.29%  114.278      51.69%

$$R^2 = 0.5847$$

$R^2$ is the percentage of variation in $Y$ explained by $X_1$, $X_2$, $X_3$ and $X_4$ together. 

What do all these coefficients of determination show? On it's own, $X_1$ (age) doesn't contribute much to the model, but when combined with $X_4$ (square footage), together they explain 47% of the variation in $Y$. $X_2$ (taxes and expenses) also contributes significantly to the model but $X_3$ (vacancy rate) does not. 

The degree of marginal linear association between $Y$ and $X_1$, when adjusted for $X_4$, increases dramatically (from 6% to 25%). 

In [52]:
42.275/168.782

0.2504710217914232

In [53]:
27.857/126.508

0.220199513074272

In [54]:
0.420/98.65

0.004257475924987328

## 7.19

### a.

As an example, here is the formula I used to transform $Y$ in Minitab:

    1/sqrt(80)*('Y - rental rate'-MeanY)/StDevY
    
The formulas for the predictor variables have the same form. The fit model is:

    Y* = -0.5479 X1* + 0.4236 X2* + 0.0485 X3* + 0.5028 X4*
    
### b.

$$b^*_2 = 0.4236$$

I suppose I can safely say that $b^*_2$ represents the change in $Y^*$ per unit change in $X_2$ while keeping all other predictor variables constant. Kutner et al. caution (p. 278) us against interpreting standardized regression coefficients as showing relative impact of predictor variables. 

### c. 

I'll be using (7.53a):

$$b_k = \frac{s_Y}{s_k}b^*_k$$

In [55]:
# I can get the descriptive stats straight from pandas
sY = cp_df['Y-rental_rate'].std()
b = {}
b['b1'] = (sY/cp_df['X1-age'].std())*-0.5479
b['b2'] = (sY/cp_df['X2-expenses'].std())*0.4236
b['b3'] = (sY/cp_df['X3-vacancy_rate'].std())*0.0485
b['b4'] = (sY/cp_df['X4-footage'].std())*0.5028
b

{'b1': -0.14204593038900654,
 'b2': 0.28198535741433894,
 'b3': 0.61983727173091874,
 'b4': 7.9249773077403444e-06}

Comparing these to the original coefficients:

    Regression Analysis: Y - rental r versus X4 - footage, X1 - age, X2 - expense, X3 - vacancy 

    Coefficients

    Term                   Coef 
    Constant             12.201 
    X4 - footage       0.000008 
    X1 - age            -0.1420 
    X2 - expenses        0.2820 
    X3 - vacancy rate      0.62 
    
Yes, these agree with the coefficients I just calculated except for slight rounding effect differences.